dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1506,19 +1513,25 @@ def median_abs_deviation(self) -> float:
             np.append([True], np.diff(bin_edges_impose) > 1e-14)
         ]
 
-        bin_counts_impose_pos: float = np.interp(
-            bin_edges_impose, bin_edges_pos, np.cumsum(np.append([0], bin_counts_pos))
+        bin_counts_impose_pos: npt.NDArray[np.float64] = np.interp(
+            bin_edges_impose,
+            bin_edges_pos,
+            np.cumsum(np.append([0], bin_counts_pos)),
+        )
+        bin_counts_impose_neg: npt.NDArray[np.float64] = np.interp(
+            bin_edges_impose,
+            bin_edges_neg,
+            np.cumsum(np.append([0], bin_counts_neg)),
         )
-        bin_counts_impose_neg: float = np.interp(
-            bin_edges_impose, bin_edges_neg, np.cumsum(np.append([0], bin_counts_neg))
+        bin_counts_impose: npt.NDArray[np.float64] = (
+            bin_counts_impose_pos + bin_counts_impose_neg
         )
-        bin_counts_impose = bin_counts_impose_pos + bin_counts_impose_neg
 
         median_inds = np.abs(bin_counts_impose - 0.5) < 1e-10
         if np.sum(median_inds) > 1:
-            return np.mean(bin_edges_impose[median_inds])
+            return cast(np.float64, np.mean(bin_edges_impose[median_inds]))
 
-        return np.interp(0.5, bin_counts_impose, bin_edges_impose)
+        return cast(np.float64, np.interp(0.5, bin_counts_impose, bin_edges_impose))
```

Comment:
how could we avoid these `casts`?

Reply:
So far I haven't found a way to remove these casts. For `np.interp`, numpy doesn't have useful type annotations for the scalar case. For `np.mean`, mypy's type annotation has a lot of overloads but most of them are typed with a return type of `Any` and it seems like our case always falls under a return `Any` case.